llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -4011,15 +4011,16 @@ KnownBits SelectionDAG::computeKnownBits(SDValue Op, const APInt &DemandedElts,
       // Fill in any known bits from range information. There are 3 types being
       // used. The results VT (same vector elt size as BitWidth), the loaded
       // MemoryVT (which may or may not be vector) and the range VTs original
-      // type. The range matadata needs the full range (i.e
+      // type. The range metadata needs the full range (i.e
       // MemoryVT().getSizeInBits()), which is truncated to the correct elt size
       // if it is know. These are then extended to the original VT sizes below.
       if (const MDNode *MD = LD->getRanges()) {
+        ConstantInt *Lower = mdconst::extract<ConstantInt>(MD->getOperand(0));
+
+        Known0 = Known0.trunc(Lower->getBitWidth());
         computeKnownBitsFromRangeMetadata(*MD, Known0);
```

Comment:
Missing test changes

Reply:
This change only affects the behavior of computeKnownBits if it is called with a vector SDValue.  There is no testcase that hits this and perhaps no call-site within LLVM that might do this.  It may require the upcoming AMDGPU changes to test.

Reply:
Do that change first then?

Reply:
> Do that change first then?

If I do the AMDGPU changes first, I will have to do it without a testcase.  The testcase needs AMDGPU and the SelectionDAG.cpp changes to pass.  I'd rather not have 3 PRs.

Reply:
Testcase added that requires SelectionDAG change without AMDGPU change.